content/inheritance/reading/cat-classes/_index.md

```diff
@@ -0,0 +1,249 @@
+{{% /notice %}}
+
+
+## `@Override`
+
+Sometimes when extending a class, we'll want to modify behavior provided by the base class. This can be done by replacing the implementation of an inherited method by a completely new method implementation. For a given method, we can do this via **method overriding**.
+
+In our example, the `noise` method of `HouseCat` overrides the method of the same name in `Cat`. When we override it, we should use the `@Override` annotation.
+
+Here are the methods in question.
+
+In `Cat`:
+
+```java
+public String noise() {
+   return "Meeeeeeooooowww!";
+}
+```
+
+In `HouseCat`:
+
+```java
+@Override
+public String noise() {
+   return "Hello, my name is " + name + "!";
+}
+```
+
+If we have a `HouseCat` object and call its `noise()` method, we will be using the method defined in `HouseCat`.
+
+```java
+Cat plainCat = new Cat(8.6);
+HouseCat cheshireCat = new HouseCat("Cheshire", 12);
+
+System.out.println(plainCat.noise()); // prints "Meeeeeeooooowww!"
+System.out.println(cheshireCat.noise()); // prints "Hello, my name is Cheshire!"
+```
+
+Similar to other Java annotations, the `@Override` annotation is not required. However, it can prevent unintentional errors and make it clear when reading your code what you intend to do. The compiler will see the annotation and check to ensure that the signatures of the base method and the overriding method match up. If they don't, it will flag an error. This can help prevent you from inadvertently creating a method with a different signature.
+
+When overriding a method from a base class, the method signatures must be exactly the same. The signature of a method includes the method name, access level, return type, and the type and number of input parameters.
+
+In addition, when overriding a method, we may call the method from the base class that we are overriding by using `super`. This calls the overridden method in the base class via `super.noise()`, carrying out the original behavior if the given conditional branch is reached.
+
+## `Object` Class
+
+In a previous lesson, we introduced the special methods `equals` and `toString`. All classes contain default implementations of these methods that can be overridden. In fact, these default methods are part of a class called `Object`. If a class does not explicitly extend another class, then it implicitly extends `Object`. So the default implementations of `equals` and `toString` (along with a few other methods) are made available to us via inheritance.
+
+Note that we should use the `@Override` annotation when we provide new implementations of these methods as well.
+
+### References:
+- [The @Override annotation (docs.oracle.com)](https://docs.oracle.com/en/java/javase/17/docs/api/index.html)
+- [The Object Class (docs.oracle.com)](https://docs.oracle.com/en/java/javase/17/docs/api/index.html)
```

Comment:
Please update links to be more specific.

Reply:
updated links